src/game/objects/pickup.c

```diff
@@ -104,3 +115,127 @@ void PickUpCollision(int16_t item_num, ITEM_INFO *lara_item, COLL_INFO *coll)
+                        g_Lara.mesh_ptrs[LM_TORSO] = g_Meshes
+                            [g_Objects[O_SHOTGUN].mesh_index + LM_TORSO];
+                    }
+                    Overlay_AddPickup(item->object_number);
+                    Inv_AddItem(item->object_number);
+                    item->status = IS_INVISIBLE;
+                    RemoveDrawnItem(item_num);
+                    g_GameInfo.stats.pickup_count++;
+                    g_Lara.interact_target.is_moving = 0;
+                }
+            }
+        } else if (g_Lara.water_status == LWS_UNDERWATER) {
+            item->pos.x_rot = -25 * PHD_DEGREE;
+
+            if ((g_Input.action && lara_item->current_anim_state == AS_TREAD
+                 && g_Lara.gun_status == LGS_ARMLESS
+                 && !g_Lara.interact_target.is_moving)
+                || (g_Lara.interact_target.is_moving
+                    && g_Lara.interact_target.item_num == item_num)) {
+
+                if (TestLaraPosition(g_PickUpBoundsUW, item, lara_item)) {
+                    if (MoveLaraPosition(
+                            &g_PickUpPositionUW, item, lara_item)) {
+                        lara_item->anim_number = AA_PICKUP_UW;
+                        lara_item->current_anim_state = AS_PICKUP;
+
+                        lara_item->goal_anim_state = AS_TREAD;
+                        lara_item->frame_number =
+                            g_Anims[lara_item->anim_number].frame_base;
+                        g_Lara.interact_target.is_moving = 0;
+                        g_Lara.gun_status = LGS_HANDSBUSY;
+                    }
+                    g_Lara.interact_target.item_num = item_num;
+                } else if (
+                    g_Lara.interact_target.is_moving
+                    && g_Lara.interact_target.item_num == item_num) {
+                    g_Lara.interact_target.is_moving = 0;
+                    g_Lara.gun_status = LGS_ARMLESS;
+                }
+            } else if (
+                g_Lara.interact_target.item_num == item_num
+                && lara_item->current_anim_state == AS_PICKUP
+                && lara_item->frame_number
+                    == g_Anims[AA_PICKUP_UW].frame_base + 18) {
+                Overlay_AddPickup(item->object_number);
+                Inv_AddItem(item->object_number);
+                item->status = IS_INVISIBLE;
+                RemoveDrawnItem(item_num);
+                g_GameInfo.stats.pickup_count++;
+                return;
+            }
+
+            if (lara_item->current_anim_state == AS_PICKUP) {
+                if (lara_item->frame_number != AF_PICKUP_UW) {
+                    return;
+                }
+                Overlay_AddPickup(item->object_number);
+                Inv_AddItem(item->object_number);
+                item->status = IS_INVISIBLE;
+                RemoveDrawnItem(item_num);
+                g_GameInfo.stats.pickup_count++;
+                return;
```

Comment:
In essence a majority of this function is a copy-paste from `PickUpCollision`. Can you merge as much of the common parts as possible into a new static function?

Reply:
I condensed `MoveLaraPosition` into `Move3DPosTo3DPos` so there's only one of those. But combining `PickUpCollision` seems like it would be a lot messier. I don't want to mess up the original pickup code.

Reply:
Can we at least merge these then?
```
            if (lara_item->frame_number == AF_PICKUP_ERASE) {
                if (item->object_number == O_SHOTGUN_ITEM) {
                    g_Lara.mesh_ptrs[LM_TORSO] = g_Meshes[g_Objects[O_SHOTGUN].mesh_index + LM_TORSO];
                }
                Overlay_AddPickup(item->object_number);
                Inv_AddItem(item->object_number);
                item->status = IS_INVISIBLE;
                RemoveDrawnItem(item_num);
                g_GameInfo.stats.pickup_count++;
                g_Lara.interact_target.is_moving = 0;
            }
```

It should be OK for `PickUpCollision` to `g_Lara.interact_target.is_moving = 0;`.

Reply:
I just realized there's probably an OG bug related to this lol. If a custom builder put a shotgun underwater, I don't think it would be added to Lara's model. I can merge the pickup item part to fix that I think.